### PR TITLE
[Eventonica] Revise Part 7 - React

### LINKED ISSUE #1421: [Eventonica] Update React portion
A child of the Eventonica epic: #1296

Modify [Old Eventonica Part 7](https://github.com/Techtonica/curriculum/blob/main/projects/eventonica/eventonica-part7-react.md) to match new reqs in README 

This should build a browser app with no API. All data should be in-memory using the code from [Part 1](https://github.com/Techtonica/curriculum/blob/main/projects/eventonica/eventonica-part1-objects.md).

A prereq for doing this part of the project is a still-unwritten React Part 4 lesson #1337

### PR DESCRIPTION
**Pull request by vegetabill**
Resolves #1421 

  - Fill in Eventonica Part 3 lesson with step-by-step examples of porting a JS DOM app to React
  - This should be rebased against #1471 first
  - Sample code adapted from @lisaau https://github.com/lisaau/Eventonica/blob/master/src/jQuery-scripts.js

### CODE REVIEW

File: projects/eventonica/eventonica-part7-react.md [hunk 1]
```diff
@@ -1,111 +1,153 @@
...
+    );
+  }
+}
+```
 
-#### Supplemental Materials
+### Re-rendering the UI in Response to Changes
 
-- Example of [calling API from React component using fetch](https://reactjs.org/docs/faq-ajax.html)
+Now, however, we want the UI to re-render when we delete the user. There is not one best way to do this so you'll have to use the lessons you learned previously to figure out how to accomplish this for the assignment.
```
Comment (alodahl) [inline projects/eventonica/eventonica-part7-react.md:149]: Isn't the best way to update the state?

Comment (alodahl) [reply projects/eventonica/eventonica-part7-react.md:149]: updating the state should always cause a re-render.

File: projects/eventonica/eventonica-part7-react.md [hunk 2]
```diff
@@ -1,111 +1,153 @@
...
-- In package.json, update the line that says `"start": "react-scripts start",` to instead say `"start": "PORT=3001 react-scripts start",`.
-  - This will make sure your React app isn't trying to run on the same port as your Express app, because your React app will now run on port 3001. Each port can only be used by one app at a time.
+You should be able to move any JavaScript files into the `public` folder and use them as-is. At this point your app should work but not really using React in any way. Make sure the basics work before proceeding.
 
-Here's an example of what `package.json` might look like now:
+### Create Your First Component
 
+Before this stage, ensure you have a commit in place with the working app.
+
+Here is example code that might be simmilar to the way you displayed a list of users in your app. Depending on how to built Part 2, your code might not be in a function like this but hopefully you can translate the concept to your own design.
```
Comment (alodahl) [inline projects/eventonica/eventonica-part7-react.md:38]:
```suggestion
Here is example code that might be similar to the way you displayed a list of users in your app. Depending on how you built Part 2, your code might not be in a function like this, but hopefully you can translate the concept to your own design.
```

File: projects/eventonica/eventonica-part7-react.md [hunk 3]
```diff
@@ -1,111 +1,153 @@
...
 
-- In your React app directory, run `npm install`.
-- Make sure your React app works by running `npm start`. You should be able to go to `http://localhost:3001/` and see it running.
-- In another Terminal tab, run your Express app. Once they are both running, you're ready to code React!
+```jsx
+// src/App.js
+import UsersList from './UsersList.js';
 
-#### Step 3: Write the React code
+const eventRecommender = new EventRecommender();
```
Comment (alodahl) [inline projects/eventonica/eventonica-part7-react.md:81]: isn't this class called Eventonica now?

File: projects/eventonica/eventonica-part7-react.md [hunk 4]
```diff
@@ -1,111 +1,153 @@
...
+eventRecommender.addUser('Lisa', 12345);
+// ... some seed data
 
-- Build out your UI! You should use React to build the UI according to your design. Take a look at `src/index.js` and `src/App.js` as starting points.
-- One of the advantages of React is reusability: think about what sorts of components you will need more than once - some examples could include styled buttons or an event info card. These are the things that you should make into components. Then all you have to do is pass in the different text or functions as props, while the rest can simply be repeated.
+class App extends React.Component {
+  render() {
+    return (
+      <div>
+        <UsersList users={eventRecommender.users} />
```
Comment (alodahl) [inline projects/eventonica/eventonica-part7-react.md:89]: eventonica

Comment (alodahl) [reply projects/eventonica/eventonica-part7-react.md:89]: theres a few more spots, too

**Review by alodahl — CHANGES_REQUESTED** [review]
Looks good, I'm just concerned about the class name, and about the re-render comment at the end.  Re-rendering on state change is like the main point of React.

**Comment by vegetabill** [comment]
Thanks for reviewing! Since I have a little cushion of time on this, going to prioritize cleaning up the whole Eventonica mess and will come back to this in a week or so.

**Comment by vegetabill** [comment]
@alodahl oof I did something dumb with git and now it thinks this is closed.

The real changes are now in #1505  including all your suggested revisions.

**Pull request closed without merging**